components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/internal/IdentitySAMLSSOServiceComponent.java

```diff
@@ -93,7 +94,16 @@ protected void activate(ComponentContext ctxt) {
             log.error(errMsg, e);
             throw new RuntimeException(errMsg, e);
         }
-
+        //register SAML ECP servlet
+        Servlet samlECPServlet = new ContextPathServletAdaptor(new SAMLECPProviderServlet(),
+                SAMLSSOConstants.SAMLECP_URL);
+        try {
+            httpService.registerServlet(SAMLSSOConstants.SAMLECP_URL, samlECPServlet, null, null);
+        } catch (Exception e) {
+            String errMsg = "Error when registering SAML ECP Servlet via the HttpService.";
+            log.error(errMsg, e);
+            throw new RuntimeException(errMsg, e);
```

Comment:
Why do we need to throw runtime exception here ?. This will retry to active this bundle with the same error

Reply:
will remove runtime exception